test/TestUtilities/Test.Utility/ZipExtensions.cs

```diff
@@ -73,8 +73,9 @@ public static void ExtractAll(this ZipArchive archive, string targetPath)
 
                 entryFullName = Uri.UnescapeDataString(entryFullName.Replace('/', Path.DirectorySeparatorChar));
 
-                var targetFile = Path.Combine(targetPath, entryFullName);
-                if (!targetFile.StartsWith(targetPath, StringComparison.OrdinalIgnoreCase))
+                string targetFile = Path.GetFullPath(Path.Combine(targetPath, entryFullName));
+                string fullDestDirPath = Path.GetFullPath(targetPath + Path.DirectorySeparatorChar);
```

Comment:
You can do this (calculate the `fullDestDirPath` outside the `foreach` loop to minimize string allocations.

Reply:
Good catch. Fixed in the latest commit.